pkg/uki/upgrade.go

```diff
@@ -74,3 +82,37 @@ func (i *UpgradeAction) Run() (err error) {
 
 	return nil
 }
+
+// installRecovery replaces the "recovery" role efi and conf files with
+// the UnassignedArtifactRole efi and loader files from dir
+func (i *UpgradeAction) installRecovery(logger v1.Logger) error {
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return fmt.Errorf("creating a tmp dir: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Dump artifact to tmp dir
+	e := elemental.NewElemental(i.cfg)
+	_, err = e.DumpSource(tmpDir, i.spec.Active.Source)
+	if err != nil {
+		return err
+	}
+
+	err = copyFile(
+		filepath.Join(tmpDir, "EFI", "kairos", UnassignedArtifactRole+".efi"),
```

Comment:
Maybe it's the review view, but where does this magic unassigned artifact role var comes from? I can't find it in this code? Is it a constant? Should it be?

Reply:
IIRC it is defined in the package as a constant

Reply:
https://github.com/kairos-io/kairos-agent/blob/58c039194f19de476cc0630a1b9350c85185e6c5/pkg/uki/common.go#L15